BostonData/lambda_function/snow_parking_intent.py

```diff
@@ -0,0 +1,205 @@
+            DRIVING_DISTANCE_TEXT_KEY: False,
+            DRIVING_TIME_TEXT_KEY: False
+        }
+
+    return closest_parking_info
+
+
+def _get_driving_info(origin, destinations):
+    """
+    Gets the driving info from the provided origin address to each destination
+    address
+
+    :param origin: string containing driving starting address
+    :param destinations: array of address to calculate driving info from origin
+    address
+    :return: dictionary with address, distance, and driving time from origin
+    address
+    """
+    url_parameters = {"origins": origin,
+                      "destinations": '|'.join(destinations),
+                      "key": GOOGLE_MAPS_API_KEY,
+                      "units": "imperial"}
+    driving_directions_url = "https://maps.googleapis.com/maps/api/" \
+                             "distancematrix/json"
+
+    driving_infos = []
+
+    with requests.Session() as session:
+        response = session.get(driving_directions_url, params=url_parameters)
+
+        if response.status_code == requests.codes.ok:
+            all_driving_data = response.json()
+            try:
+                driving_data_row = all_driving_data["rows"][0]
+                for (driving_data, address) in \
+                        zip(driving_data_row["elements"], destinations):
+                    try:
+                        driving_info = {
+                            DRIVING_DISTANCE_VALUE_KEY:
+                                driving_data["distance"]["value"],
+                            DRIVING_DISTANCE_TEXT_KEY:
+                                driving_data["distance"]["text"],
+                            DRIVING_TIME_VALUE_KEY:
+                                driving_data["duration"]["value"],
+                            DRIVING_TIME_TEXT_KEY:
+                                driving_data["duration"]["text"],
+                            PARKING_LOCATION_KEY: address}
+                        driving_infos.append(driving_info)
+                    except KeyError:
+                        print("Could not parse driving info {}"
+                              .format(driving_data))
+            except KeyError:
+                return driving_infos
+        else:
+            print("Failed to get driving directions")
+
+    return driving_infos
+
+
+def _get_emergency_parking_data():
+    """
+    Gets the emergency parking info from Boston Data
+
+    :return: array of emergency parking info as provided in the Boston data
+    csv file
+    """
+    parking_data = []
+    with requests.Session() as session:
+        url = "http://bostonopendata-boston.opendata.arcgis.com/" + \
+              "datasets/53ebc23fcc654111b642f70e61c63852_0.csv"
+        response = session.get(url)
+
+        if response.status_code == requests.codes.ok:
+            response_data = response.content
+            csv_reader = csv.reader(response_data.splitlines())
+            parking_data = list(csv_reader)

```

Comment:
this line is throwing an error "_csv.Error: iterator should return strings, not bytes (did you open the file in text mode?)" I can't tell why that is the case.